libraries/AP_Baro/AP_Baro.h

```diff
@@ -251,7 +254,7 @@ class AP_Baro
     bool init_done;
 
     uint8_t msp_instance_mask;
-
+    AP_Vector3f pos_offset;
```

Comment:
```suggestion
#if AP_BARO_POSITION_COMPENSATION_ENABLED
    AP_Vector3f pos_offset;
#endif  // AP_BARO_POSITION_COMPENSATION_ENABLED
```

... and around the new methods and whatever else is being introduced.  THis should be a no-compiler-output change if the feature is not enabled.  You only do this in the main code, not in the SITL stuff.